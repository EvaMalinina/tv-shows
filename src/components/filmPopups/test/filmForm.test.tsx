import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import FilmPopup from "../FilmForm";

describe('It should render FilmForm component', () => {
  const mockStore = configureStore();
  const filmMockProps = {
    movieId: '123456',
    name: 'Film',
    desc: 'I am film',
    category: 'comedy',
    year: 2021,
    img: 'link',
    runtime: 100
  }
  const initialState = {};
  const typeAdd = 'add';

  const popupsReducer = {
      add: true
    }

  const singleMovieReducer = {
      movie: {}
    }

  const alertsReducer = {
    alertsVisible: false
  }
  
  const store = mockStore({
      popupsReducer,
      singleMovieReducer,
      alertsReducer
    })
  

console.log('------store', store.getState())
  const FilmAddPopupComponent = (type: any) => {
    return <>{
      <Provider store={store}>
        <FilmPopup type={type}/>
      </Provider>
    }</>
  }

  test('Render add movie popup', () => {
    const { asFragment } = render(FilmAddPopupComponent(typeAdd));
    expect(asFragment()).toMatchSnapshot();
  });

  // test('Render edit movie popup', () => {
  //   const { asFragment } = render(FilmAddPopupComponent(typeAdd));
  //   expect(asFragment()).toMatchSnapshot();
  // });
});