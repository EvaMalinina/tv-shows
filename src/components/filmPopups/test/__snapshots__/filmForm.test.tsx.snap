// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render FilmForm component Render add movie popup 1`] = `
<DocumentFragment>
  <div
    class="sc-furvIG kPEIxN"
  >
    <div
      class="sc-pVTma dRHGXU"
    >
      <div
        class="sc-bdvvaa fdicDB"
      >
        <form
          class="sc-jrQzUz iYvNTz"
        >
          <h2>
            add movie
          </h2>
          <button
            class="sc-kDThTU glIxWQ"
          >
            âœ•
          </button>
          <label
            for="title"
          >
            Title
            <input
              id="title"
              name="title"
              value=""
            />
          </label>
          <label
            for="startDate"
          >
            Release Date
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </label>
          <label
            for="movie-url"
          >
            movie url
            <input
              id="movie-url"
              name="movieUrl"
              value=""
            />
          </label>
          <label
            for="genre"
          >
            genre
            <div
              style="margin-top: 5px;"
            >
              <div
                class=" css-2b097c-container"
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-idbc9a-control"
                >
                  <div
                    class=" css-xcyr2j-ValueContainer"
                  >
                    <div
                      class=" css-5sz5u5-singleValue"
                    />
                    <input
                      aria-autocomplete="list"
                      class="css-wmatm6-dummyInput-DummyInput"
                      id="react-select-2-input"
                      readonly=""
                      tabindex="0"
                      value=""
                    />
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </label>
          <label
            for="overview"
          >
            Overview
            <input
              id="overview"
              name="overview"
              value=""
            />
          </label>
          <label
            for="runtime"
          >
            Runtime
            <input
              id="runtime"
              name="runtime"
              value="60"
            />
          </label>
          <div
            class="sc-iqsfdx gJBEqk"
          >
            <button
              class="sc-gKckTs sc-crHlIS bZSJUX iyEYTv"
            >
              Reset
            </button>
            <button
              class="sc-gKckTs sc-egiSv bZSJUX HOFr"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
